packages/core/src/aad.ts

```diff
@@ -0,0 +1,253 @@
+   * Calls getAllAccounts and determines the correct account to sign into,
+   * currently defaults to first account found in cache.
+   *
+   * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/docs/Accounts.md
+   */
+  getAccount(): AccountInfo | null {
+    const currentAccounts = this.msalApplication.getAllAccounts();
+    if (currentAccounts && currentAccounts.length >= 1) {
+      return currentAccounts[0];
+    }
+
+    return null;
+  }
+
+  /**
+   * Gets the token to read user profile data silently, or falls back to interactive redirect.
+   */
+  async getProfileTokenRedirect(): Promise<void | AuthenticationResult> {
+    return this.getTokenRedirect(
+      this.silentProfileRequest,
+      this.profileRedirectRequest
+    );
+  }
+
+  /**
+   * Checks whether we are in the middle of a redirect and handles state accordingly.
+   * Only required for redirect flows.
+   *
+   * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#redirect-apis
+   */
+  async initAuth(): Promise<AccountInfo | void> {
+    const account = await this.handleAuthRedirect();
+
+    return this.handleAuthAccountResult(account);
+  }
+
+  /**
+   * Calls loginPopup or loginRedirect based on given signInType.
+   * @param signInType
+   */
+  async login(
+    signInType: AzureADSingInType = LOGIN_REDIRECT
+  ): Promise<AccountInfo | void> {
+    if (signInType === LOGIN_POPUP) {
+      try {
+        const { account } = await this.msalApplication.loginPopup(
+          this.loginRequest
+        );
+
+        return this.handleAuthAccountResult(account);
+      } catch (error) {
+        console.error(error);
+      }
+    } else {
+      await this.msalApplication.loginRedirect(this.loginRedirectRequest);
+    }
+  }
+
+  /**
+   * Logs out of current account.
+   */
+  async logout(): Promise<void> {
+    const logOutRequest: EndSessionRequest = {
+      account: this.account || undefined,
+    };
+
+    await this.msalApplication.logout(logOutRequest);
+
+    this.account = undefined;
+  }
+
+  /**
+   * Returns already acquired access token
+   */
+  async getCDFToken(): Promise<string | null> {
+    if (!this.account) return null;
+
+    const response: AuthenticationResult = await this.msalApplication.acquireTokenSilent(
+      this.silentCDFTokenRequest
+    );
+
+    return response.accessToken;
+  }
+
+  private getCDFScopes(): string[] {
+    return [
+      `https://${this.cluster}.cognitedata.com/user_impersonation`,
+      `https://${this.cluster}.cognitedata.com/IDENTITY`,
+    ];
+  }
+
+  private get loginRequest(): PopupRequest {
+    return {
+      scopes: this.userScopes,
+      extraScopesToConsent: this.getCDFScopes(),
+    };
+  }
+
+  private get loginRedirectRequest(): RedirectRequest {
+    return {
+      scopes: this.userScopes,
+      extraScopesToConsent: this.getCDFScopes(),
+      redirectStartPage: window.location.origin,
```

Comment:
Set this to `window.location.href`. This won't do anything without setting `navigateToLoginRequestUrl` to true in auth options.